src/NzbDrone.Core/HealthCheck/Checks/DownloadClientSortingCheck.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+using NLog;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Datastore.Events;
+using NzbDrone.Core.Download;
+using NzbDrone.Core.Download.Clients;
+using NzbDrone.Core.Localization;
+using NzbDrone.Core.RemotePathMappings;
+using NzbDrone.Core.RootFolders;
+using NzbDrone.Core.ThingiProvider.Events;
+
+namespace NzbDrone.Core.HealthCheck.Checks
+{
+    [CheckOn(typeof(ProviderUpdatedEvent<IDownloadClient>))]
+    [CheckOn(typeof(ProviderDeletedEvent<IDownloadClient>))]
+    [CheckOn(typeof(ModelEvent<RootFolder>))]
+    [CheckOn(typeof(ModelEvent<RemotePathMapping>))]
+
+    public class DownloadClientSortingCheck : HealthCheckBase
+    {
+        private readonly IProvideDownloadClient _downloadClientProvider;
+        private readonly Logger _logger;
+
+        public DownloadClientSortingCheck(IProvideDownloadClient downloadClientProvider,
+                                      Logger logger,
+                                      ILocalizationService localizationService)
+            : base(localizationService)
+        {
+            _downloadClientProvider = downloadClientProvider;
+            _logger = logger;
+        }
+
+        public override HealthCheck Check()
+        {
+            var clients = _downloadClientProvider.GetDownloadClients();
+
+            foreach (var client in clients)
+            {
+                try
+                {
+                    var status = client.GetStatus();
```

Comment:
Can we add clientName back in here so that if there are changes upstream they are more likely to cherry-pick in without conflicts please